executor/newdeploy/newdeploy.go

```diff
@@ -508,3 +508,19 @@ func (deploy *NewDeploy) waitForDeploy(depl *v1beta1.Deployment, replicas int32)
 	}
 	return nil, errors.New("failed to create deployment within timeout window")
 }
+
+func (deploy *NewDeploy) cleanupNewdeploy(name string, ns string) {
+	log.Printf("Error creating new deploy function resources, deleting all resources with name %v, in namespace %v", name, ns)
```

Comment:
Log the error where the error happens, instead of here.  That way you can log the error message.  It also keeps this function more generic.  Also suggest returning a multierror from this function even if its not used in the error code path.

Also, `fnDelete` should be changed to call this function, since it features the exact same sequence of delete calls.